File: Shiski-Backend/src/main/java/com/bootcamp/feature/carrito/service/CarritoService.java
package com.bootcamp.feature.carrito.service;

import com.bootcamp.feature.carrito.dto.CarritoRequest;
import com.bootcamp.feature.carrito.dto.CarritoResponse;

public interface CarritoService {
    CarritoResponse agregarProducto(CarritoRequest request);
    CarritoResponse obtenerCarrito(Long id);
    CarritoResponse actualizarCantidad(CarritoRequest carritoRequest);
        // 1. Lógica para buscar el producto en el carrito del usuario.
        // 2. Actualizar la cantidad del producto.
        // 3. Guardar el carrito actualizado en el repositorio.
        // 4. Mapear la entidad actualizada a CarritoResponse y devolverla.


    void vaciarCarrito(Long usuarioId);
        // 1. Lógica para buscar el carrito del usuario por su ID.
        // 2. Eliminar todos los productos del carrito.
        // 3. Guardar los cambios en el repositorio.

    void eliminarProducto(Long id);
}

File: Shiski-front-react/src/modules/api/services/carritoService.js
import axiosInstance from '../config/axiosConfig';

const CARRITO_URL = '/api/v1/carrito';

const carritoService = {
  // Obtener carrito del usuario
  obtenerCarrito: async (usuarioId) => {
    try {
      const response = await axiosInstance.get(`${CARRITO_URL}/${usuarioId}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  // Agregar producto al carrito
  agregarProducto: async (carritoData) => {
    try {
      const response = await axiosInstance.post(CARRITO_URL, carritoData);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  // Actualizar cantidad de producto
  actualizarCantidad: async (carritoData) => {
    try {
      const response = await axiosInstance.put(CARRITO_URL, carritoData);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  // Eliminar producto del carrito
  eliminarProducto: async (carritoId) => {
    try {
      await axiosInstance.delete(`${CARRITO_URL}/${carritoId}`);
    } catch (error) {
      throw error;
    }
  },

  // Vaciar carrito completo
  vaciarCarrito: async (usuarioId) => {
    try {
      await axiosInstance.delete(`${CARRITO_URL}/vaciar/${usuarioId}`);
    } catch (error) {
      throw error;
    }
  }
};

export default carritoService;